scripts/bump_deps.lua

```diff
@@ -0,0 +1,378 @@
+	end
+	return rv
+end
+
+-- Run a command, return nil on failure
+local function run(cmd)
+	return _run(cmd, false, "")
+end
+
+-- Run a command, die on failure with err_msg
+local function run_die(cmd, err_msg)
+	return _run(cmd, true, err_msg)
+end
+
+local function check_executable(cmd)
+	local cmd_path = run({ "command", "-v", cmd })
+	if cmd_path == nil then
+		return false
+	end
+	local rv = run({ "test", "-x", cmd_path })
+	return rv ~= nil
+end
+
+local function require_executable(cmd)
+	local cmd_path = run_die({ "command", "-v", cmd }, cmd .. " not found!")
+	run_die({ "test", "-x", cmd_path }, cmd .. " is not executable")
+end
+
+local function rm_file_if_present(path_to_file)
+	run({ "rm", "-f", path_to_file })
+end
+
+local nvim_src_dir = vim.fn.getcwd()
+local temp_dir = nvim_src_dir .. "/tmp"
+run({ "mkdir", "-p", temp_dir })
+local gh_res_path = temp_dir .. "/gh_res.json"
+
+local function get_dependency(dependency_name)
+	local dependency_table = {
+		["LuaJIT"] = {
+			repo = "LuaJIT/LuaJIT",
+			symbol = "LUAJIT",
+		},
+		["libuv"] = {
+			repo = "libuv/libuv",
+			symbol = "LIBUV",
+		},
+		["Luv"] = {
+			repo = "luvit/luv",
+			symbol = "LUV",
+		},
+		["tree-sitter"] = {
+			repo = "tree-sitter/tree-sitter",
+			symbol = "TREESITTER",
+		},
+	}
+	local dependency = dependency_table[dependency_name]
+	if dependency == nil then
+		p("Not a dependency: " .. dependency_name)
+		die()
+	end
+	dependency.name = dependency_name
+	return dependency
+end
+
+local function dl_gh_ref_info(repo, ref)
+	require_executable("curl")
+
+	rm_file_if_present(gh_res_path)
+	local ref_status_code = run_die({
+		"curl",
+		"-s",
+		"-H",
+		"'Accept: application/vnd.github.v3+json'",
+		"-w",
+		"'%{http_code}'",
+		"https://api.github.com/repos/" .. repo .. "/commits/" .. ref,
+		"-o",
+		gh_res_path,
+	}, "Failed to fetch commit details from GitHub")
+	if ref_status_code ~= "'200'" then
+		p("Not a valid ref: " .. ref)
+		die()
+	end
+end
+
+local function get_json_field(jsonfile_path, field)
+	require_executable("jq")
+	return run({ "jq", "-r", "." .. field, jsonfile_path })
+end
+
+local function get_archive_info(repo, ref)
+	require_executable("curl")
+
+	local archive_name = ref .. ".tar.gz"
+	local archive_path = temp_dir .. "/" .. archive_name
+	local archive_url = "https://github.com/" .. repo .. "/archive/" .. archive_name
+
+	rm_file_if_present(archive_path)
+	run_die({ "curl", "-sL", archive_url, "-o", archive_path }, "Failed to download archive from GitHub")
+
+	local archive_sha = run({ "sha256sum", archive_path }):gmatch("%w+")()
+	return { url = archive_url, sha = archive_sha }
+end
+
+local function write_cmakelists_line(symbol, kind, value)
+	require_executable("sed")
+
+	local cmakelists_path = nvim_src_dir .. "/" .. "third-party/CMakeLists.txt"
+	run_die({
+		"sed",
+		"-i",
+		"-e",
+		"s/set(" .. symbol .. "_" .. kind .. ".*$" .. "/set(" .. symbol .. "_" .. kind .. " " .. value .. ")" .. "/",
+		cmakelists_path,
```

Comment:
You need to be careful about the version of `sed`, the in-place flag `-i` requires the GNU version, which isn't available on macOS/BSD by default.

I think It might be best to try to extract this part completely to a new cmake file that gets generated as a template, but maybe that's out of scope for this PR, although I believe it would be easier to maintain and reason about (alá lockfile/package.json)

Reply:
`sed -i` is available on macOS.

Reply:
`man sed` on my machine.

Reply:
Did you try the command as specified here?

Reply:
Also, your macOS is outdated.